### PR TITLE
many: preparations for image code to fetch model prereqs

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: image/image.go [hunk 1]
```diff
@@ -279,48 +273,31 @@ func copyLocalSnapFile(snapName, targetDir string) (copyiedSnapFn string, info *
 }
 
 type downloadOptions struct {
```
Comment (mvo5) [inline image/image.go]: Nice how this got simpler!

File: image/image.go [hunk 2]
```diff
@@ -279,48 +273,31 @@ func copyLocalSnapFile(snapName, targetDir string) (copyiedSnapFn string, info *
...
 	Download(name string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (path string, err error)
 }
 
-func downloadSnapWithSideInfo(name string, opts *downloadOptions) (targetPath string, info *snap.Info, err error) {
+func downloadSnapWithSideInfo(sto Store, name string, opts *downloadOptions) (targetPath string, info *snap.Info, err error) {
 	if opts == nil {
 		opts = &downloadOptions{}
 	}
 
-	// FIXME: avoid global mutation
```
Comment (mvo5) [inline image/image.go]: \o/ !!!

**Comment by mvo5** [comment]
Very nice work, thank you!

**Comment by chipaca** [comment]
:+1:!

**Pull request merged**